### PR TITLE
Modules

### PR DESCRIPTION
**Pull request by Juan-M-V**
Divide program into module and main

### CODE REVIEW

File: src/hangman.rs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        Hangman { 
+            word, 
+            attempts: 0, 
+            misses: Default::default(), 
+            hits: vec![SYMBOL_LETTER_TO_COMPLETE; vec_size],
+        }
+    }
+
+    fn guess_letter(&mut self, l:char) -> AttemptResult {
+
```
Comment (Loco1982) [inline src/hangman.rs]: This line is not necessary. Please remove it.

File: src/hangman.rs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    fn guess_letter(&mut self, l:char) -> AttemptResult {
+        
+        if self.hits.contains(&l) || self.misses.contains(&l) {
+            return AttemptResult::CanKeepPlaying;
+        }
+
+        let mut guessed_letter = false;
+        let mut missing_letters = false;
+
+        for (i, c) in self.word.chars().enumerate(){
```
Comment (Loco1982) [inline src/hangman.rs]: These variables should have a descriptive name. Please change it.

File: src/hangman.rs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        let vec_size = word.len();
+        Hangman { 
+            word, 
+            attempts: 0, 
+            misses: Default::default(), 
+            hits: vec![SYMBOL_LETTER_TO_COMPLETE; vec_size],
+        }
+    }
+
+    fn guess_letter(&mut self, l:char) -> AttemptResult {
```
Comment (Loco1982) [inline src/hangman.rs]: Please change the variable to a descriptive name.

File: src/hangman.rs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        let mut guessed_vec:Vec<char> = Vec::<char>::new();
+
+        for l in &self.hits {
+            guessed_vec.push(*l);
+            guessed_vec.push(' ');
+        }
+        guessed_vec.pop();
+
+        let guessed:String = guessed_vec.into_iter().collect();
+
```
Comment (Loco1982) [inline src/hangman.rs:80]: This space is not necessary, please remove it.

File: src/hangman.rs [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        let len_misses = self.misses.len();
+        let mut misses_vec = vec![' '; len_misses * 2 - 1];
+
+        for (i, l) in self.misses[..len_misses - 1].into_iter().enumerate() {
+            misses_vec[i * 2] = *l; 
+        }
+
+        let misses:String = misses_vec.into_iter().collect();
+
+
```
Comment (Loco1982) [inline src/hangman.rs]: This space is not necessary, please remove it.

**Review by Loco1982 — CHANGES_REQUESTED** [review]
There are some formal things to correct.

File: src/hangman.rs (hunk 1 above)
Comment (Juan-M-V) [reply src/hangman.rs]: Removed in next commit

File: src/hangman.rs (hunk 2 above)
Comment (Juan-M-V) [reply src/hangman.rs]: Changed c to letter_from_word but left i as the index

File: src/hangman.rs (hunk 3 above)
Comment (Juan-M-V) [reply src/hangman.rs]: Changed l to letter

File: src/hangman.rs (hunk 4 above)
Comment (Juan-M-V) [reply src/hangman.rs:80]: Removed

File: src/hangman.rs (hunk 5 above)
Comment (Juan-M-V) [reply src/hangman.rs]: Removed

**Pull request merged**